nextjs-template/src/components/presentation/deviceEventInfo.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable import/prefer-default-export */
+import {
+  Device,
+  DeviceEnvVars,
+  Event,
+  FleetEnvVars,
+} from "../../services/DomainModel";
+
+// function to combine devices with their event data and env vars
+export function getNormalizedDeviceData(
+  devices: Device[],
+  deviceEvents: Event[],
+  deviceEnvVars: DeviceEnvVars[],
+  fleetEnvVars: FleetEnvVars[]
+) {
+  const deviceEventInfo = devices.map((device) => {
+    // consider also filtering out alarm.qo events as well to keep the data cleaner and more uniform
+    const filterEventsByDevice = deviceEvents
+      .filter((event) => event.deviceUID.deviceUID === device.id.deviceUID)
```

Comment:
You're correct - the `IDBuilder.ts` file is the one that puts the type and id together for a project, device, event. It could be more than is necessary, but it's easy to tell at a glance what's being reference when objects contain more than one type of ID. For now, I just went with the pattern